packages/tanstack-react-query/test/queryKeyable.test.tsx

```diff
@@ -82,10 +118,42 @@ describe('get queryFilter', () => {
           }
         `);
 
-      return 'heck';
+      return 'some text';
     }
 
-    ctx.renderApp(<Heck />);
+    ctx.renderApp(<Component />);
+  });
+
+  test('type inference for query filters', async () => {
+    await using ctx = testContext();
+
+    const { useTRPC } = ctx;
+
+    function Component() {
+      const trpc = useTRPC();
+      const query = useQueryClient();
+
+      const a = trpc.bluesky.post.byId.queryFilter(
+        { id: '1' },
+        {
+          predicate(query) {
+            const data = query.setData('__result');
+            assertType<'__result' | undefined>(data);
+
+            return true;
+          },
+        },
+      );
+
+      // Fix the lint error and typescript complains, the lint error is wrong
+      // eslint-disable-next-line @typescript-eslint/no-unnecessary-type-assertion
```

Comment:
Not sure why it's complaining here, tslint is wrong. Fix the warning and typescript complains instead